### PR TITLE
Fix USER_NOT_FOUND exception on line 417

### PR DESCRIPTION
**Pull request by AnvaySingh**
## What does this PR do?

This PR fixes issue #4727 regarding Bug Report: Get User Returns 500 if User Does Not Exist

## Test Plan

Execute the 'Get User' API with a User ID that does not exist and check for the error code displayed. It should be not 500

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/4727

### Have you added your change to the [Changelog](https://github.com/appwrite/appwrite/blob/master/CHANGES.md)?

Not yet

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by stnguyen90** [comment]
Would you please comment on the issue so we can assign it to you?

**Comment by stnguyen90** [comment]
The correct response is returned:

image

**Comment by stnguyen90** [comment]
@AnvaySingh, actually, would you please add a test case in:

https://github.com/appwrite/appwrite/blob/de8552e9018df460d07bdbd587716b95e0c9f083/tests/e2e/Services/Users/UsersBase.php#L612-L648

The end, before `return $data` should include a test for failure like:

https://github.com/appwrite/appwrite/blob/de8552e9018df460d07bdbd587716b95e0c9f083/tests/e2e/Services/Users/UsersBase.php#L596-L606

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
@AnvaySingh, actually, would you please add a test case in:

https://github.com/appwrite/appwrite/blob/de8552e9018df460d07bdbd587716b95e0c9f083/tests/e2e/Services/Users/UsersBase.php#L612-L648

The end, before `return $data` should include a test for failure like:

https://github.com/appwrite/appwrite/blob/de8552e9018df460d07bdbd587716b95e0c9f083/tests/e2e/Services/Users/UsersBase.php#L596-L606

File: tests/e2e/Services/Users/UsersBase.php [hunk 1]
```diff
@@ -644,6 +644,19 @@ public function testGetUser(array $data): array
...
 
+        
+        /** 
+          * Test for FAILURE 
+          */ 
+         $response = $this->client->call(Client::METHOD_GET, '/users' . $data['userId'], array_merge([ 
+             'content-type' => 'application/json', 
+             'x-appwrite-project' => $this->getProject()['$id'], 
+         ], $this->getHeaders()), [ 
+             'queries' => ['cursorAfter("unknown")']
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:655]: No queries for this request

File: tests/e2e/Services/Users/UsersBase.php [hunk 2]
```diff
@@ -644,6 +644,19 @@ public function testGetUser(array $data): array
...
+          * Test for FAILURE 
+          */ 
+         $response = $this->client->call(Client::METHOD_GET, '/users' . $data['userId'], array_merge([ 
+             'content-type' => 'application/json', 
+             'x-appwrite-project' => $this->getProject()['$id'], 
+         ], $this->getHeaders()), [ 
+             'queries' => ['cursorAfter("unknown")'] 
+         ]); 
+  
+         $this->assertEquals(400, $response['headers']['status-code']);
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:658]: Shouldn't the expected error code be 404?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: tests/e2e/Services/Users/UsersBase.php (hunk 2 above)
Comment (AnvaySingh) [reply tests/e2e/Services/Users/UsersBase.php]: Updated the error code here

File: tests/e2e/Services/Users/UsersBase.php [hunk 3]
```diff
@@ -644,6 +644,17 @@ public function testGetUser(array $data): array
         $this->assertIsInt($users['body']['total']);
         $this->assertGreaterThan(0, $users['body']['total']);
 
+        
+        /** 
+          * Test for FAILURE 
+          */ 
+         $response = $this->client->call(Client::METHOD_GET, '/users' . $data['userId'], array_merge([
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:651]: The ID passed in here should be for an ID that doesn't exist so that it returns 404.

**Comment by stnguyen90** [comment]
Closing as already resolved.

**Pull request closed without merging**